### PR TITLE
Feat:(issue_1958) Add support for multiple layouts to TimestampFlag

### PR DESCRIPTION
**Pull request by horockey**
Resolves [issue 1958](https://github.com/urfave/cli/issues/1958)

### CODE REVIEW

File: makeout.txt
Comment (bartekpacia) [inline makeout.txt:1]: what's this file?

Comment (horockey) [reply makeout.txt:1]: Forgot to ignore. Removed

File: flag_test.go [hunk 1]
```diff
@@ -2284,9 +2284,37 @@ func TestTimestampFlagApply(t *testing.T) {
...
+	for timeStr, expectedRes := range map[string]time.Time{
+		now.Format(time.DateTime): now.Truncate(time.Second),
+		now.Format(time.TimeOnly): now.Truncate(time.Second),
+		now.Format(time.Kitchen):  now.Truncate(time.Minute),
+	} {
+		err := set.Parse([]string{"--time", timeStr})
+		assert.NoError(t, err)
+		assert.Equal(t, expectedRes, set.Lookup("time").Value.(flag.Getter).Get())
+	}
+}
```
Comment (Juneezee) [inline flag_test.go:2313]: Because the layout is now changed from `Layout string` to `AvailableLayouts []string`, we need to test more edge cases such as

1. A `nil` `AvailableLayouts`
2. An empty `AvailableLayouts`
3. Combination of invalid and valid layouts
4. All invalid layouts
5. etc.

You could use [Table-driven tests style](https://go.dev/wiki/TableDrivenTests).

**Review by Juneezee — COMMENTED** [review]
Thanks for your PR! 

Awaiting for more maintainers feedback

File: flag_test.go (hunk 1 above)
Comment (horockey) [reply flag_test.go]: Thanks for the feedback. New tests have been added

File: flag_timestamp.go [hunk 2]
```diff
@@ -9,16 +10,16 @@ type TimestampFlag = FlagBase[time.Time, TimestampConfig, timestampValue]
 
 // TimestampConfig defines the config for timestamp flags
 type TimestampConfig struct {
-	Timezone *time.Location
-	Layout   string
+	Timezone         *time.Location
+	AvailableLayouts []string
```
Comment (dearchap) [inline flag_timestamp.go:14]:
```suggestion
	Layouts []string
```

File: flag_timestamp.go [hunk 3]
```diff
@@ -9,16 +10,16 @@ type TimestampFlag = FlagBase[time.Time, TimestampConfig, timestampValue]
...
 
 // timestampValue wrap to satisfy golang's flag interface.
 type timestampValue struct {
-	timestamp  *time.Time
-	hasBeenSet bool
-	layout     string
-	location   *time.Location
+	timestamp        *time.Time
+	hasBeenSet       bool
+	availableLayouts []string
```
Comment (dearchap) [inline flag_timestamp.go:21]:
```suggestion
	layouts []string
```

File: flag_timestamp.go [hunk 4]
```diff
@@ -53,16 +54,67 @@ func (t *timestampValue) Set(value string) error {
...
+	}
+
+	for _, layout := range t.availableLayouts {
+		var locErr error
+
+		timestamp, locErr = time.ParseInLocation(layout, value, t.location)
+		// TODO: replace with errors.Join() after upgrading to go 1.20 or newer
+		// OR use external error wrapping, if acceptable
+		if locErr != nil {
+			if err == nil {
```
Comment (dearchap) [inline flag_timestamp.go:72]: Use the multiError  https://github.com/urfave/cli/blob/main/errors.go#L29

File: flag_timestamp.go [hunk 5]
```diff
@@ -11,7 +11,10 @@ type TimestampFlag = FlagBase[time.Time, TimestampConfig, timestampValue]
 // TimestampConfig defines the config for timestamp flags
 type TimestampConfig struct {
 	Timezone *time.Location
-	Layouts  []string
+	// Available layouts for flag value.
+	//
+	// Note that value for formats with missing year/date will be interpreted as current year/date respectively.
+	Layouts []string
```
Comment (bartekpacia) [inline flag_timestamp.go:17]: I'd appreciate an example of what is a correct layout, or a link to docs.

Comment (horockey) [reply flag_timestamp.go:17]: Definition of a correct layout stays the same (comparing to main) - it is such a string `lo`, that causes no error calling `time.Parse(lo, val)`, where `val` is time, formatted same as `lo`.
Valid format: `15:04`
Invalid format: `11:51`
Playground example: https://go.dev/play/p/EW82fbXJdPg
More about timestamp layouts [HERE](https://pkg.go.dev/time#pkg-constants)

Comment (bartekpacia) [reply flag_timestamp.go:17]: Please include this in the doc comment.

Comment (horockey) [reply flag_timestamp.go]: Done

**Review by dearchap — APPROVED** [review]
Thank you @horockey
